### PR TITLE
Update WordPress Configuration and Extending

### PR DESCRIPTION
**Pull request by joshcanhelp**
This PR makes changes to the Configuration and Extending sections of the CMS > WordPress documentation. Specifically:

- Add documentation for the `auth0_settings_constant_prefix` hook
- Add information on settings using PHP constants
- Correct the setting names, order, and descriptions

This can be reviewed now but needs to wait until the release next week to be merged. 

Thank you!

### CODE REVIEW

File: articles/cms/wordpress/configuration.md [hunk 1]
```diff
@@ -139,53 +139,80 @@ See our [dedicated page on Social Connections](/identityproviders#social) for de
 
 1. Scroll down and click **Save Changes**.
 
+## PHP Constant Setting Storage
+
+Plugin settings can be saved to the database (default) or they can be set using a specifically-name PHP constant defined before plugins are loaded (typically in the `wp-config.php` file). This will allow for sensitive data like the client secret, API token, and migration token to be stored more securely (assuming that file they are defined in is [stored securely](https://codex.wordpress.org/Hardening_WordPress)).
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: `specifically named` maybe?

File: articles/cms/wordpress/configuration.md [hunk 2]
```diff
@@ -139,53 +139,80 @@ See our [dedicated page on Social Connections](/identityproviders#social) for de
...
+The PHP constants are defined like so:
+
+```php
+// Storing the client secret in a constant.
+// Add to wp-config.php or anywhere before WP_Auth0_Options_Generic is instantiated.
+// Constant name is "AUTH0_ENV_" + option key in caps.
+define( 'AUTH0_ENV_CLIENT_SECRET', 'YOUR_CLIENT_SECRET_HERE' );
+```
+
+The default constant name should be `AUTH0_ENV_` followed by the option name to override in all caps (the prefix can be modified with the `auth0_settings_constant_prefix` filter [explained here](https://auth0.com/docs/cms/wordpress/extending#auth0_settings_constant_prefix)). All plugin options that can be overridden and their keys can be found in the `WP_Auth0_Options::defaults()` [method](https://github.com/auth0/wp-auth0/blob/master/lib/WP_Auth0_Lock_Options.php).
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: Should use relative URL: `/cms/wordpress/extending#auth0_settings_constant_prefix`

File: articles/cms/wordpress/configuration.md [hunk 3]
```diff
@@ -139,53 +139,80 @@ See our [dedicated page on Social Connections](/identityproviders#social) for de
...
+
 ## Plugin settings
 
 ### Basic
 
-* **Domain:** The app Domain copied from the Application settings in your dashboard.
+* **Domain:** The domain Domain copied from the Application settings in your dashboard.
 
-* **Client ID:** The app Client ID copied from the Application settings in your dashboard.
+* **Custom Domain:** The Custom Domain for your tenant, if one is configured. More information on Custom Domains [here](https://auth0.com/docs/custom-domains).
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: Same here, `/custom-domains`.

File: articles/cms/wordpress/configuration.md [hunk 4]
```diff
@@ -246,17 +273,15 @@ social Connections from working properly.
...
-
-* **Extra settings:** A valid JSON object that includes options to call Lock with. This overrides all other options set above. For a list of available options, see [Lock: User configurable options](/libraries/lock/customization) (e.g.: `{"disableResetAction": true }`).
+* **Extra Settings:** A valid JSON object that includes options to call Lock with. This overrides all other options set above. For a list of available options, see [Lock: User configurable options](/libraries/lock/customization) (e.g.: `{"disableResetAction": true }`).
 
-* **Twitter consumer key and consumer secret:** The credentials from your Twitter app. For instructions on creating an app on Twitter, see [Obtain Consumer and Secret Keys for Twitter](/connections/social/twitter).
+* **Custom signup fields:** This field is the JSON that describes the custom signup fields for lock. It should be a valid json and allows the use of functions (for validation). [More info here](/libraries/lock/v10/new-features#custom-sign-up-fields).
```
Comment (jeffreylees) [inline articles/cms/wordpress/configuration.md]: `* **Custom signup fields:** This field is the JSON that describes the custom signup fields for Lock. It should be valid JSON and allows the use of functions (for validation). You can find more information in the Lock documentation on [custom signup fields](/libraries/lock/v11/configuration#additionalsignupfields-array-).`

Changing a few copywriting items + the doc you were linking to is an old "what's new in Lock 10" doc, and could be a confusing one to be wandering around in + may be removed soon.

**Review by jeffreylees — CHANGES_REQUESTED** [review]

**Comment by joshcanhelp** [comment]
@jeffreylees - thank you for the review! Made the changes suggested and added "Option names" to each of the configuration fields just now. This one is ready to merge once you're happy with it, release going out now 👍

**Review by jeffreylees — APPROVED** [review]

**Pull request merged**